### ISSUE TITLE
Book search is failing to reindex correctly if events are changed (add/delete/reassociate) without updating work record

### ISSUE THREAD

**Issue by clmahoney** [Princeton-CDH/mep-django#667]
**Describe the bug**
When a work is altered on the test-admin site, the test-production site adds the new work but does not reindex the old work on the book sort page. For example, when I divided "Vanity Fair" from one periodical with 16 events to the periodical with 4 events and another entry for the novel Vanity Fair with 12 events, the production site shows the novel entry with 12 events but still shows the periodical work with 16 entries when it should only be 4. This is only wonky on the search page; when you click through to the individual work bibliography page, circulation page, and card page it does the expected 4 events. 
RSK put it more concisely on Slack:  the error is happening if you change events (especially add/delete/reassociate) without updating the work record.

**To reproduce**
1) Change events for a work without altering the work entry itself 
2) See that the altered work does not show up with its changes in the book search page 

**Expected behavior**
When events within a work are changed (added, deleted, or re-associated) the book search page should show the correct number of events. 

**Screenshots**
Example of error with Vanity Fair: 
![Screen Shot 2020-06-13 at 10 54 43 PM](https://user-images.githubusercontent.com/16673934/84583734-eb8adf00-adc9-11ea-99d7-a8203f5654a1.png)

**Comment by rlskoeser** [comment]
Have configured a nightly cron job to reindex works in the interim until this is resolved. The cron job should be turned off when this is fixed.

**Comment by jkotin** [comment]
I went to the test admin site and created a new item -- _2020_ by Virginia Woolf. I created one event -- a Hemingway borrow. I then went to the test site: the book was there, but the search page was showing 0 events.

I added another borrow to the same book -- by Stein -- and still 0 events on the search page.  I don't know how I tested it differently earlier. 

I added a borrow to a preexisting item -- Brooke's 1914. The search page only shows one event when there should be two.
